partitioner/src/main/java/io/confluent/connect/storage/partitioner/Partitioner.java

```diff
@@ -30,8 +30,25 @@
 public interface Partitioner<T> {
   void configure(Map<String, Object> config);
 
+  /**
+   * Returns string representing the output path for a sinkRecord to be encoded and stored.
+   * @param sinkRecord The record to be stored by the Sink Connector
+   * @return The path/filename the SinkRecord will be stored into after it is encoded
```

Comment:
Add:
```
     * @deprecated use {@link #encodePartition(SinkRecord, long)} instead
```
and then add the annotation:
```
@Deprecated
```

Reply:
I'd push back on this -- There are legitimate cases for not passing in the "now" time (e.g. a field-pased partitioner). Deprecating these creates a ton of warnings/failures because there are indeed places where it should be fine to call the old behavior (incl in the default function of the interface).